proto/canto/csr/v1/tx.proto

```diff
@@ -2,12 +2,97 @@ syntax = "proto3";
 package canto.csr.v1;
 
 // this line is used by starport scaffolding # proto/tx/import
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
 
 option go_package = "github.com/Canto-Network/Canto/v2/x/csr/types";
 
-// Msg defines the Msg service.
+// Msg defines the csr Msg service.
 service Msg {
-    // this line is used by starport scaffolding # proto/tx/rpc
+    // register a new CSR with a set of smart contracts
+    rpc RegisterCSR(MsgRegisterCSR) returns (MsgRegisterCSRResponse) {
+        option (google.api.http).get = "/canto/csr/v1/register_csr";
+    };
+    // update an existing CSR with a new set of smart contracts
+    rpc UpdateCSR(MsgUpdateCSR) returns (MsgUpdateCSRResponse) {
+        option (google.api.http).get = "/canto/csr/v1/update_csr";
+    };
+    // withdraw rewards for CSRs
+    rpc WithdrawCSR(MsgWithdrawCSR) returns (MsgWithdrawCSRResponse) {
+        option (google.api.http).get = "/canto/csr/v1/withdraw_csr";
+    };
 }
 
-// this line is used by starport scaffolding # proto/tx/message
+// Wrapper so that we can create a map to an array of Nonces in MsgRegisterCSR and MsgUpdateCSR
+message NoncesArray {
```

Comment:
Try removing wrappers around types? This makes it so that we have to unwrap the decoded values to use them as we would use the object if it was just a repeated repeated uint

Reply:
ok will look into this